src/raw_node.rs

```diff
@@ -419,11 +419,20 @@ impl<T: Storage> RawNode<T> {
     /// Panics if passed with the context of context.can_async() == false
     pub fn on_entries_fetched(&mut self, context: GetEntriesContext) {
         match context.0 {
-            GetEntriesFor::SendAppend { to, aggressively } => {
+            GetEntriesFor::SendAppend {
+                to,
+                term,
+                aggressively,
+            } => {
                 if self.raft.prs().get(to).is_none() {
                     // the peer has been removed, do nothing
                     return;
                 }
+                if self.raft.term != term || self.raft.state != StateRole::Leader {
```

Comment:
Should be checked before L427. Actually, role changes implies term changes.

Reply:
Oh, seems quorum active check will make it step down without changing its term. In this case, it should also not send any entries to other peers. Should add a case to cover that.